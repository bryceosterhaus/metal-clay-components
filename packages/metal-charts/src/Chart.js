'use strict';

import Soy from 'metal-soy';

import ChartBase from './ChartBase';
import templates from './Chart.soy.js';

/**
 * Metal Chart component.
 */
class Chart extends ChartBase {
}
Soy.register(Chart, templates);

export { Chart };
export default Chart;
