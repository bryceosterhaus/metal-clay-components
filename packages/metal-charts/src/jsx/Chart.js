'use strict';

import {JSXComponent} from 'metal-jsx';

import ChartBase from '../ChartBase';

/**
 * Metal Chart component for use with JSX.
 */
class ChartJSX extends JSXComponent {
	render() {
		return <div class="metal-chart"></div>;
	}
}

Object.assign(ChartJSX.prototype, ChartBase);
ChartJSX.PROPS = ChartBase.STATE;

ChartJSX.prototype.getStateObj_ = function() {
	return this.props;
};

export {ChartJSX};
export default ChartJSX;
