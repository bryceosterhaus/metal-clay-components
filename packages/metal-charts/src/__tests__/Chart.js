'use strict';

import {bb} from 'billboard.js';

import Chart from '../Chart';

beforeAll(() => {
	jest.spyOn(bb, 'generate').mockImplementation();
});

beforeEach(() => {
	bb.generate.mockReset();
});

afterAll(() => {
	bb.generate.mockReset();
	bb.generate.mockRestore();
});

describe('Chart', function() {
	it('should be pass correctly formatted configuration options to billboard.js', () => {
		new Chart({
			columns: []
		});

		const config = JSON.stringify(bb.generate.mock.calls[0][0]);

		expect(config).toMatchSnapshot();
	});

	it('should format columns array into billboard.js compatible data', () => {
		new Chart({
			columns: [
				{
					axis: 'y',
					class: 'data1',
					color: 'red',
					data: [1, 2, 3],
					id: 'data1',
					name: 'Data 1',
					regions: [{
						end: 2,
						start: 1,
						style: 'dashed'
					}],
					type: 'line',
					xs: 'x1'
				},
				{
					axis: 'y2',
					class: 'data2',
					color: 'blue',
					data: [1, 2, 3],
					id: 'data2',
					name: 'Data 2',
					regions: [{
						end: 3,
						start: 2,
						style: 'dashed'
					}],
					type: 'spline',
					xs: 'x2'
				}
			]
		});

		const config = JSON.stringify(bb.generate.mock.calls[0][0]);

		expect(config).toMatchSnapshot();
	});
});
