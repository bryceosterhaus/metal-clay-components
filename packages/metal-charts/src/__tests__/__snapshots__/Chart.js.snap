// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should be pass correctly formatted configuration options to billboard.js 1`] = `"{\\"axis\\":{},\\"bindto\\":{},\\"data\\":{\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[]},\\"zoom\\":{}}"`;

exports[`Chart should format columns array into billboard.js compatible data 1`] = `"{\\"axis\\":{},\\"bindto\\":{},\\"data\\":{\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[[\\"data1\\",1,2,3],[\\"data2\\",1,2,3]],\\"axes\\":{\\"data1\\":\\"y\\",\\"data2\\":\\"y2\\"},\\"classes\\":{\\"data1\\":\\"data1\\",\\"data2\\":\\"data2\\"},\\"colors\\":{\\"data1\\":\\"red\\",\\"data2\\":\\"blue\\"},\\"names\\":{\\"data1\\":\\"Data 1\\",\\"data2\\":\\"Data 2\\"},\\"regions\\":{\\"data1\\":[{\\"end\\":2,\\"start\\":1,\\"style\\":\\"dashed\\"}],\\"data2\\":[{\\"end\\":3,\\"start\\":2,\\"style\\":\\"dashed\\"}]},\\"types\\":{\\"data1\\":\\"line\\",\\"data2\\":\\"spline\\"},\\"xs\\":{\\"data1\\":\\"x1\\",\\"data2\\":\\"x2\\"}},\\"zoom\\":{}}"`;

exports[`Chart should rerender chart when new columns are passed 1`] = `"{\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[[\\"data1\\",1,2,3,4]]}"`;

exports[`Chart should unload removed columns when new columns are passed 1`] = `"{\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[[\\"data2\\",1,2,3]],\\"unload\\":[\\"data1\\"]}"`;
